src/test/java/cucumber_tests/support/ScenarioContext.java

```diff
@@ -0,0 +1,28 @@
+package cucumber_tests.support;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/***
+ *  ScenarioContext helps to store data and pass it other steps
+ */
+public class ScenarioContext {
```

Comment:
Nice idea! I would recommend to use singleton for this solution to avoid losing data
Also, there might be issues with parallel execution with the current implementation, just keep it in mind for the task with parallel execution.

Reply:
fixed